FrontEnd/FrontEnd.Web/Components/Pages/Recipes/Form.razor

```diff
@@ -198,23 +198,45 @@
         public string? Value { get; set; }
     }
 
-    private void HandleValidSubmit()
+    private async Task HandleValidSubmitAsync()
     {
-        var recipeDto = new global::Recipes.Public.Recipe
+        if (InitialData == null)
+        {
+            // Creation
+            var requestDto = new global::Recipes.Public.RecipeCreateDTO
+            {
+                Title = FormData.Title!,
+                Description = FormData.Description,
+                CookingTime = TimeSpan.FromMinutes((double)FormData.CookingTimeInMinutes!),
+                Servings = FormData.Servings!.Value,
+                Ingredients = FormData.Ingredients.Select(x => x.Value!).ToList(),
+                Instructions = FormData.Instructions.Select(x => x.Value!).ToList(),
+                Tags = FormData.Tags.ToList(),
+                Image = FormData.Image
+            };
+            
+            var result = await RecipeService.CreateAsync(requestDto);
+            NavigationManager.NavigateTo($"recipes/{result.Id}");
+        }
+        else
         {
-            Title = FormData.Title!,
-            AuthorId = 1,
-            Description = FormData.Description,
-            CookingTime = TimeSpan.FromMinutes((double)FormData.CookingTimeInMinutes!),
-            Servings = FormData.Servings!.Value,
-            Ingredients = FormData.Ingredients.Select(x => x.Value!),
-            Instructions = FormData.Instructions.Select(x => x.Value!),
-            Image = FormData.Image
-        };
-
-        // TODO: Send out `recipeDto` to the Recipes API
-        // If FormData.Id is null, send to create
-        // otherwise send to update recipe with id FormData.Id
+            // Update
+            var requestDto = new global::Recipes.Public.RecipeUpdateDTO
+            {
+                Title = FormData.Title!,
+                Description = FormData.Description,
+                CookingTime = TimeSpan.FromMinutes((double)FormData.CookingTimeInMinutes!),
+                Servings = FormData.Servings!.Value,
+                Ingredients = FormData.Ingredients.Select(x => x.Value!).ToList(),
+                Instructions = FormData.Instructions.Select(x => x.Value!).ToList(),
+                Tags = FormData.Tags.ToList(),
+                Image = FormData.Image,
+                Version = InitialData.Version
```

Comment:
Shouldn't update change version?

Reply:
It shouldn't?
Upon opening the edit page, the recipe is loaded - it gives us the latest version. When updating, we try to supply what we think is the latest version.

Reply:
Oops, I had previously missed the opt locking PR. I can see it now, right.